aten/src/ATen/native/cpu/SortingKernel.cpp

```diff
@@ -179,47 +194,52 @@ static void sort_kernel(
     // https://github.com/pytorch/pytorch/issues/91420
     return;
   }
+
+  AT_DISPATCH_V2(indices.scalar_type(), "sort_kernel", AT_WRAP([&] {
 #ifdef USE_FBGEMM
-  if (can_use_radix_sort(values, descending)) {
-    parallel_sort1d_kernel(values, indices);
-    return;
-  }
+    if (can_use_radix_sort(values, descending)) {
+      using signed_t = UnsignedType<scalar_t>::signed_type; //FBGEMM explicit template instantiations don't contain unsigned type.
+      auto* const indices_ptr = reinterpret_cast<signed_t *>(indices.data_ptr<scalar_t>());
+      parallel_sort1d_kernel<signed_t>(values, indices_ptr);
```

Comment:
Explicit template instantiation in fbgemm::radix_sort_parallel only supports uint8_t among unsigned dtype(s). That's why unsigned values should be reinterpret_casted to signed values.